cli/src/main/kotlin/cli/util/Util.kt

```diff
@@ -20,35 +22,35 @@ fun generateAST(
     parser: Parser,
     args: Map<String, String>,
 ): AST {
-    // Get the file path from the command line arguments
-    val filePath = getFilePath(args)
-    // Read the file into a string
-    val file = File(filePath).readText()
-
-    // Lex the file
+    val file = File(getFilePath(args)).readText()
     val tokens = lexer.lex(file)
+    return parseTokens(parser, tokens)
+}
 
-    // Parse the tokens and return the AST
-    when (val parserResult = parser.parse(tokens, 0)) {
-        is SuccessResult -> {
-            return parserResult.value
-        }
-        is FailureResult -> {
-            // TODO: Add the exact position, not the index of the last token
-            throw IllegalArgumentException(
-                "Parsing Error: ${parserResult.message}",
-            )
-        }
-    }
+fun generateBufferedAST(
+    lexer: LineLexer,
+    parser: Parser,
+    args: Map<String, String>,
+): AST {
+    val file = File(getFilePath(args)).readText()
```

Comment:
I think this would not be buffered, since it is reading the whole file at once. Take a look at the utils class for reading files as tokens, maybe you could use that one.